### PR TITLE
[common] supports configuring force-lookup.

### PR DESCRIPTION
**Pull request by liming30**
### Purpose

Linked issue: subtask of #3868 

[common] supports configuring force-lookup

### Tests

- `org.apache.paimon.table.PrimaryKeyFileStoreTableTest#testForceLookupCompaction`

### CODE REVIEW

**Comment by JingsongLi** [comment]
Comment in Issue.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/AsyncLookupSinkWrite.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
 
 /**
  * {@link StoreSinkWrite} for tables with lookup changelog producer and {@link
- * org.apache.paimon.CoreOptions#CHANGELOG_PRODUCER_LOOKUP_WAIT} set to false.
+ * org.apache.paimon.CoreOptions#LOOKUP_WAIT} set to false.
  */
 public class AsyncLookupSinkWrite extends StoreSinkWriteImpl {
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/AsyncLookupSinkWrite.java:39]: You don't need to modify Flink lookup join? Can you explain which specific case has changed after introducing `force-lookup`?

Comment (liming30) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/AsyncLookupSinkWrite.java:39]: I plan to put the modification of `FileStoreLookupFunction` into the next issue. This pr mainly introduces `force-lookup` to change compaction. 

The next pr will modify the validation of `streamingReadingValidate` and modify `FullCacheLookupTable` to implement streaming consumption data. There is almost no need to modify `PrimaryKeyPartialLookupTable`.

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java [hunk 2]
```diff
@@ -2024,6 +2032,10 @@ public boolean metadataIcebergCompatible() {
         return options.get(METADATA_ICEBERG_COMPATIBLE);
     }
 
+    public boolean forLookup() {
```
[the remark this answers is not in the document]
Comment (liming30) [reply paimon-common/src/main/java/org/apache/paimon/CoreOptions.java]: Resolved.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/AsyncLookupSinkWrite.java (hunk 1 above)
Comment (JingsongLi) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/AsyncLookupSinkWrite.java:39]: OK

**Comment by JingsongLi** [comment]
+1

**Pull request merged**